lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -1738,41 +1739,48 @@ selectCoins
     -> ApiT WalletId
     -> ApiSelectCoinsPayments n
     -> Handler (ApiCoinSelection n)
-selectCoins ctx@ApiLayer {..} genChange (ApiT wid) body = do
-    let md = body ^? #metadata . traverse . #getApiT
-    withWorkerCtx ctx wid liftE liftE $ \wrk -> do
-        let db = wrk ^. dbLayer
-        era <- liftIO $ NW.currentNodeEra netLayer
-        wdrl <- case body ^. #withdrawal of
-            Nothing -> pure NoWithdrawal
-            Just apiWdrl ->
-                shelleyOnlyMkWithdrawal @s @k @n @'CredFromKeyK
-                    netLayer txLayer db wid era apiWdrl
-        let outs = addressAmountToTxOut <$> body ^. #payments
+selectCoins ctx@ApiLayer {..} argGenChange (ApiT walletId) body = do
+    era <- liftIO $ NW.currentNodeEra netLayer
+    recentEra@(AnyRecentEra (_ :: WriteTx.RecentEra e)) <- guardIsRecentEra era
```

Comment:
Just wondering: can't the `W.build...` functions be polymorphic in `era` instead of taking the `AnyRecentEra`?

Reply:
Actually, you're right - as recent era check has moved up the stack (HTTP API layer) the existentially captured recent era in not required by the W.buildTransaction (its enough to have a `IsRecentEra =>` constraint) ! Removed.